src/risk_atlas_nexus/blocks/risk_mapping/risk_mapper.py

```diff
@@ -90,7 +95,7 @@ def generate(self, new_risks: List[Risk], existing_risks: List[Risk], inference_
                 result = re.search('ID:(.*), Name', s)
                 result2 = re.search('Name:(.*), Description:', s)
                 
-                mapping = Mapping(subject_id= self._format_with_curie(new_prefix, nr.id), 
+                mapping = Mapping(subject_id= self._format_with_curie(nr.isDefinedByTaxonomy, nr.id), 
                                   subject_label=nr.name, 
                                   predicate_id=self._bucket_semantic_score(uid),
                                   object_id=self._format_with_curie(taxonomy_for_mapping[result.group(1).strip()], result.group(1)),
```

Comment:
The key to `taxonomy_for_mapping`  is an risk_id (Line no. 80).

Is it right to access this dictionary with a risk name? I think `result.group(1)` carries a risk name, right?